src/pages/profile-page/avatar/avatar.js

```diff
@@ -33,42 +32,57 @@ const Avatar = ({ setUserImageUrl, userImageUrl, setUpload, setDeleteAvatar, t }
     return false;
   };
 
-  const checkDimension = async (result) => {
-    const image = new Image();
-
-    const promiseImg = new Promise((resolve, reject) => {
-      image.onload = () => {
-        if (image.height <= 100 && image.width <= 100) {
-          resolve(true);
-        } else {
-          resolve(false);
+  const checkOrResizeImageTo100x100 = (base64Str, maxWidth = 200, maxHeight = 200) => new Promise((resolve) => {
+    const originalImage = new Image();
+    originalImage.src = base64Str;
+    originalImage.onload = () => {
+      const canvas = document.createElement('canvas');
+      const canvasCtx = canvas.getContext('2d');
+      let width = originalImage.naturalWidth;
+      let height = originalImage.naturalHeight;
+      if (width >= height) {
+        if (width > maxWidth) {
+          height *= maxWidth / width;
+          width = maxWidth;
         }
-      };
-
-      image.onerror = (error) => reject(error);
-    });
-
-    image.src = result;
-
-    return promiseImg;
+      } else if (height >= maxHeight) {
+        width *= maxHeight / height;
+        height = maxHeight;
+      }
+      canvas.width = width;
+      canvas.height = height;
+      canvasCtx.drawImage(originalImage, 0, 0, width, height);
+      resolve(canvas.toDataURL('image/jpeg', 1));
+    };
+  });
+
+  const dataURItoBlob = (dataURI) => {
```

Comment:
Same here.

Reply:
general functions have been moved to utils folder.